fili-core/src/main/java/com/yahoo/bard/webservice/web/ResponseWriterSelector.java

```diff
@@ -0,0 +1,19 @@
+// Copyright 2017 Oath Inc.
+// Licensed under the terms of the Apache license. Please see LICENSE.md file distributed with this work for terms.
+package com.yahoo.bard.webservice.web;
+
+
+/**
+ * Response writer selector object.
+ */
+public interface ResponseWriterSelector {
+    /**
+     * Select ResponseWriter given certain type of format from DataApiRequest.
+     *
+     * @param request  ApiRequest object with all the associated info in it
+     * @return  Writer of given format type
+     */
+    default ResponseWriter select(DataApiRequest request) {
```

Comment:
This should absolutely not be defaulted. In my opinion, default implementations should only be used for methods that are added to preserve backwards compatibility, or maybe as convenience methods for invoking the other, meatier methods on the interface.

We definitely shouldn't be using a default implementation that relies on one of the implementations of the interface.